frontend/catalyst/jax_tracer.py

```diff
@@ -859,6 +874,15 @@ def trace_quantum_measurements(
 
     for i, o in enumerate(outputs):
         if isinstance(o, MeasurementProcess):
+
+            # Check if the measurement is supported shot-vector where num_of_total_copies > 1
+            num_of_total_copies = get_device_total_shot_copies(device)
+            if num_of_total_copies > 1 and o.return_type.value != "sample":  # qml.sample()
+                raise NotImplementedError(
```

Comment:
Nice!

Looks like this check is missing test coverage however. Typically we verify errors as well using pytest.

Reply:
Good catch! I'll add test coverage for this check. Thanks for pointing that out!